plugins/syntax/numeral.ml

```diff
@@ -78,7 +78,9 @@ let locate_int63 () =
   else None
 
 let has_type f ty =
-  let (sigma, env) = Pfedit.get_current_context () in
+  (* This shouldn't take the context from the proof *)
```

Comment:
Really? And also not the evarmap?

Reply:
No idea what to do here, I've reworked the code a bit so it seems clear where the things do come from, but indeed that's a vernacular declaration so in fact it should be forbidden to be used in the proof, moreover with the problem with the state this introduces.

[numeral notations have state outside the summary [hihger order] and this breaks for example jscoq in 8.9]

Reply:
See updated file, we could thread the state would we wanted but indeed that part of the code is veyr problematic :S